prov/gni/src/gnix_cq.c

```diff
@@ -541,8 +548,42 @@ DIRECT_FN STATIC ssize_t gnix_cq_readerr(struct fid_cq *cq,
 	}
 
 	event = container_of(entry, struct gnix_cq_entry, item);
-
-	memcpy(buf, event->the_entry, sizeof(struct fi_cq_err_entry));
+	gnix_cq_err = event->the_entry;
+
+	buf->op_context = gnix_cq_err->op_context;
+	buf->flags = gnix_cq_err->flags;
+	buf->len = gnix_cq_err->len;
+	buf->buf = gnix_cq_err->buf;
+	buf->data = gnix_cq_err->data;
+	buf->tag = gnix_cq_err->tag;
+	buf->olen = gnix_cq_err->olen;
+	buf->err = gnix_cq_err->err;
+	buf->prov_errno = gnix_cq_err->prov_errno;
+
+	if (gnix_cq_err->err_data != NULL) {
+#if 0
```

Comment:
Yes, but add a comment here that we should do version checks once we decide how we're going to do them.